server/src/main/java/org/opensearch/gateway/ClusterStateUpdaters.java

```diff
@@ -120,7 +121,19 @@ static ClusterState updateRoutingTable(final ClusterState state) {
         // initialize all index routing tables as empty
         final RoutingTable.Builder routingTableBuilder = RoutingTable.builder(state.routingTable());
         for (final IndexMetadata cursor : state.metadata().indices().values()) {
-            routingTableBuilder.addAsRecovery(cursor);
+            // Whether IndexMetadata is recovered from local disk or remote it doesn't matter to us at this point.
```

Comment:
When I restart any one node in the cluster, will this flow be called?

Reply:
This would be called only when cluster state recovery is performed. It wont be called on normal node restart

Reply:
can we add an assert?